ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirestoreDataStore.kt

```diff
@@ -75,8 +77,15 @@ internal constructor(
     Firebase.messaging.subscribeToTopic(surveyId).await()
   }
 
-  /** Calls Cloud Function to refresh the current user's profile info in the remote database. */
+  /**
+   * Calls Cloud Function to refresh the current user's profile info in the remote database if
+   * network is available.
+   */
   override suspend fun refreshUserProfile() {
+    if (!networkManager.isNetworkAvailable()) {
```

Comment:
The RemoteDataStore just abstracts calls to the remote server, in this case Firestore. Deciding whether to do this or not (i.e., if online) should be up to the calling repository imo, not the data store.

Reply:
Done